### DISCUSSION TITLE
Problem adapting the tokenizer to split on number

### DISCUSSION THREAD

**Discussion by lnatspacy** [explosion/spaCy#8949] [Help: Coding & Implementations] [answered]
Hi, 

I'm having to customize the tokenizer for a German NER model. I've already added suffixes and infixes but I'm having trouble figuring out this particular problem. 
I have tokens that end in a number e.g. "Recht3". I'd like this to be 2 tokens i.e. "Recht" and "3". At the same time, I don't want to split other tokens that end in a number like for example 2010.

I could ad this suffix (?<=[^0-9])[0-9]$ but technically, suffixes are only supposed to be used for punctuation, right?
How do I best accomplish this?
I guess I could use retokenize but I'd have to iterate through all tokens to find such tokens(?) which would be a real drag on performance. 

Is there another way ?

**Comment by adrianeboyd** [comment, marked as answer]
You can use suffixes for whatever you want! That kind of pattern would be fine, there are also similar patterns in the defaults for `"."`:

https://github.com/explosion/spaCy/blob/8448c7dbc54f18a3283f8218a72c57987e4c43a8/spacy/lang/punctuation.py#L27-L28

**Reply by lnatspacy** [reply]
Ok, got it - Thanks :)
